Student/student.js

```diff
@@ -0,0 +1,98 @@
+let studentArray;
+let studentIdToDelete;
+let dialog = document.getElementById('dialogId')
+let errDialog = document.getElementById("errorDialog")
+let studentDetailsPopup = document.getElementById("displayStudentDetailsId")
+let addDialog = document.getElementById("addStuDia")
+let deleteStudent = (event) => {
+    studentIdToDelete = event.target.id;
+    dialog.showModal()
+}
+
+let removeStudent = () => {
+
+    fetch("/api/deletestudent", {
+        method: "DELETE"
+    })
+        .then(res => res.json())
+        .then(res => {
+            if (res.status == "success") {
+                document.getElementById(studentIdToDelete).parentElement.remove()
+                dialog.close()
+            }
+        }).catch(err => {
+            dialog.close()
+            errDialog.showModal()
+        })
+
+}
+let closeErrorDialog = () => {
+    errDialog.close()
+}
+
+let closeDialog = () => {
+    dialog.close()
+}
+
+
+let popupStudent = (event) => {
+    document.getElementById("details").innerHTML = event.target.innerHTML
+    studentDetailsPopup.showModal()
+}
+
+let closeDialogStudentDEtails = () => {
+    studentDetailsPopup.close()
+}
+
+
+fetch("/api/students")
+    .then(res => res.json())
+    .then(res => {
+        studentArray = res;
+
+        studentArray.forEach(student => {
+            let studentListDiv = document.createElement('div')
```

Comment:
Why do you need extra `div` here ?

Reply:
because to separate li and button. It is required while showing student details. Otherwise if we keep everything in li, button also will be shown in the popup.